llvm/lib/Target/AMDGPU/FLATInstructions.td

```diff
@@ -1826,6 +1826,21 @@ multiclass GlobalFLATStorePats<FLAT_Pseudo inst, SDPatternOperator node,
   }
 }
 
+class LoadIntrinWithScopeMetadata<SDPatternOperator intrin> : PatFrag<
+  (ops node:$ptr),
+  (intrin $ptr, srcvalue)>;
+def global_load_b128_intrin_pat : LoadIntrinWithScopeMetadata<int_amdgcn_global_load_b128>;
+
+class StoreIntrinWithScopeMetadata<SDPatternOperator intrin> : PatFrag<
+  (ops node:$data, node:$ptr),
+  (intrin $ptr, $data, srcvalue)>;
+def global_store_b128_intrin_pat : StoreIntrinWithScopeMetadata<int_amdgcn_global_store_b128>;
```

Comment:
Since there's only one use, you can get rid of the intermediate StoreIntrinWithScopeMetadata/LoadIntrinWithScopeMetadata classes